mdoc/Mono.Documentation/Updater/DocUtils.cs

```diff
@@ -412,7 +412,11 @@ public static void AppendFieldValue(StringBuilder buf, FieldDefinition field)
                     buf.Append(" = ").Append(val.ToString());
                 else if (val is IFormattable)
                 {
-                    string value = ((IFormattable)val).ToString(null, CultureInfo.InvariantCulture);
+                    string value = "";
+                    if (field.FieldType.FullName == "System.Double")
+                    { value = ((IFormattable)val).ToString("R", CultureInfo.InvariantCulture); }

```

Comment:
I don't think using `R` is necessarily appropriate.

On Full Framework, there is no guarantee that `R` provides a string that will actually roundtrip (due to issues in the floating-point parser that can't be fixed due to compat reasons, etc; for .NET Core 3.0 and later, `R` and `G` will always roundtrip).

The appropriate thing is to likely use `G17` which will produce enough digits to roundtrip. `17` is the maximum precision supported by full framework (.NET core supports up to `99`). `G17` will always produce a roundtrippable string on .NET Core and will in almost every scenario produce a roundtrippable string on full framework.

`G9` is the precision that should be used for `System.Single`

Reply:
I don't know about the guarantees of Mono; but I believe it is sharing the parsing/formatting code with .NET Core; so it likely depends on the version of Mono being used.